src/pypdfium2/_library_scope.py

```diff
@@ -3,21 +3,70 @@
 
 import sys
 import atexit
+import ctypes
+import os
+from typing import List, Optional
 import pypdfium2.raw as pdfium_c
 import pypdfium2.internal as pdfium_i
 
 
-def init_lib():
+def _create_font_path_array(font_paths: List[str]):
+    """
+    Create a null-terminated array of C strings from font paths for PDFium.
+    Returns a pointer that can be used for FPDF_LIBRARY_CONFIG.m_pUserFontPaths.
+    """
+    if not font_paths:
+        return None
+
+    # Convert paths to bytes and add null terminator
+    font_paths_bytes = [path.encode('utf-8') for path in font_paths]
+    font_paths_bytes.append(None)  # Null terminator
+
+    # Create array of c_char_p
+    char_array = (ctypes.c_char_p * len(font_paths_bytes))(*font_paths_bytes)
+
+    # Get the correct type from FPDF_LIBRARY_CONFIG structure
+    font_path_type = pdfium_c.FPDF_LIBRARY_CONFIG._fields_[1][1]  # m_pUserFontPaths type
+
+    # Cast to the expected type
+    return ctypes.cast(char_array, font_path_type)
+
+
+def init_lib(font_paths: Optional[List[str]] = None):
+    """
+    Initialize PDFium library with optional custom font paths.
+
+    Args:
+        font_paths: List of paths to font directories or font files that PDFium should use.
+                   These will be used instead of or in addition to system fonts.
+    """
     assert not pdfium_i.LIBRARY_AVAILABLE
     if pdfium_i.DEBUG_AUTOCLOSE:  # pragma: no cover
         # FIXME not shown on auto-init, because DEBUG_AUTOCLOSE can only be set on the caller side after pypdfium2 has been imported...
         print("Initialize PDFium", file=sys.stderr)
-    
+
+    # Handle font paths
+    if font_paths:
+        # Validate font paths exist
+        valid_paths = []
+        for path in font_paths:
+            if os.path.exists(path):
+                valid_paths.append(path)
+            else:
+                if pdfium_i.DEBUG_AUTOCLOSE:  # pragma: no cover
+                    print(f"Warning: Font path does not exist: {path}", file=sys.stderr)
+
+        font_paths_ptr = _create_font_path_array(valid_paths)
+        if pdfium_i.DEBUG_AUTOCLOSE and valid_paths:  # pragma: no cover
+            print(f"Loading {len(valid_paths)} custom font paths: {valid_paths}", file=sys.stderr)
```

Comment:
This isn't related to auto-closing so it should not be controlled by the `DEBUG_AUTOCLOSE` condition, and it should use standard, controllable logging means, not print. Or if it were about auto-closing, it should use the `_safe_debug` function, not print.

Same above.